staging/src/k8s.io/apiserver/pkg/server/options/server_run_options.go

```diff
@@ -336,6 +356,16 @@ func (s *ServerRunOptions) AddUniversalFlags(fs *pflag.FlagSet) {
 	fs.DurationVar(&s.ShutdownWatchTerminationGracePeriod, "shutdown-watch-termination-grace-period", s.ShutdownWatchTerminationGracePeriod, ""+
 		"This option, if set, represents the maximum amount of grace period the apiserver will wait "+
 		"for active watch request(s) to drain during the graceful server shutdown window.")
+}
 
-	utilfeature.DefaultMutableFeatureGate.AddFlag(fs)
+// Complete fills missing fields with defaults.
+func (s *ServerRunOptions) Complete() error {
+	if s.FeatureGate == nil {
+		s.FeatureGate = utilfeature.DefaultFeatureGate
+	}
+	if s.EffectiveVersion == nil {
```

Comment:
Can we disallow `nil` in validation instead. We're requiring during construction (correctly I think), so it will like be more friendly to make it clear: no really, supply this.

Reply:
> Can we disallow `nil` in validation instead. We're requiring during construction (correctly I think), so it will like be more friendly to make it clear: no really, supply this.

This still looks like a mistake.